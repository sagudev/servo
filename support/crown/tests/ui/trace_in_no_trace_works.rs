/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/. */

/// Fake jstraceable
pub trait JSTraceable {}

#[crown::trace_in_no_trace_lint::must_not_have_traceable]
struct NoTrace<T>(T);

struct Bar;
impl JSTraceable for Bar {}

struct Foo(NoTrace<Bar>);

fn main() {}
