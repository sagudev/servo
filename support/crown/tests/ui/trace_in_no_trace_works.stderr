error: must_not_have_traceable marked wrapper must not have jsmanaged inside on 0-th position. Consider removing the wrapper.
  --> $DIR/trace_in_no_trace_works.rs:10:12
   |
10 | struct Foo(NoTrace<Bar>);
   |            ^^^^^^^^^^^^
   |
   = note: `#[deny(crown::trace_in_no_trace)]` on by default

error: aborting due to previous error

